test/library/run-options.test.js

```diff
@@ -98,6 +98,31 @@ describe('Newman run options', function () {
         });
     });
 
+    describe('bail modifiers', function () {
+        it('should skip collection run in case of error when folder is specified', function (done) {
+            newman.run({
+                collection: 'test/fixtures/run/failed-request.json',
+                bail: ['folder']
+            }, function (err) {
+                expect(err).to.be.ok();
+                expect(err.message).to.be('getaddrinfo ENOTFOUND 123.random.z 123.random.z:443');
+
+                done();
+            });
+        });
+
+        it('should gracefully stop a collection run in case of error with no additional modifiers', function (done) {
```

Comment:
Also add a test where both `folder` and `failure` are present as `bail` options.